kamu-core/src/domain/repos/metadata_chain.rs

```diff
@@ -395,17 +395,61 @@ pub struct RefCASError {
 #[derive(Error, Debug)]
 pub enum AppendValidationError {
     #[error(transparent)]
-    HashMismatch(HashMismatchError),
+    HashMismatch(#[from] HashMismatchError),
     #[error("First block has to be a seed, perhaps new block does not link to the previous")]
     FirstBlockMustBeSeed,
     #[error("Attempt to append seed block to a non-empty chain")]
     AppendingSeedBlockToNonEmptyChain,
     #[error("Invalid previous block")]
-    PrevBlockNotFound(BlockNotFoundError),
+    PrevBlockNotFound(#[from] BlockNotFoundError),
     #[error(transparent)]
-    SequenceIntegrity(SequenceIntegrityError),
+    SequenceIntegrity(#[from] SequenceIntegrityError),
     #[error("System time has to be monotonically non-decreasing")]
     SystemTimeIsNotMonotonic,
+    #[error("Watermark has to be monotonically increasing")]
+    WatermarkIsNotMonotonic,
```

Comment:
I guess it also has to be a point <= Current time

Reply:
Not necessarily. For example if some clerical procedure requires data entry once a month and all data is available on 15th day of the month - it might be useful to set the watermark at 31st to signify to consumers that we do not anticipate any more data this month.